dev-requirements.txt

```diff
@@ -8,3 +8,6 @@ pre-commit == 2.20.0
 pytest == 7.2.0
 pytest-cov == 4.0.0
 requests-mock == 1.11.0
+pyiceberg @ git+https://github.com/apache/iceberg#subdirectory=python
+pydantic == 2.0
+ray[data] ~= 2.6
```

Comment:
`ray[data]` has conflicting versions in `dev-requirements` and `requirements.txt`. Was it intended?

Reply:
Yes, this is intentional for now (see follow-up issue noted in description to revert this behavior later once Ray creates an official release that is compatible with Pydantic 2.x). The different dependency versions stated in `dev-requirements`  intentionally override `requirements` so that `make test-integration` passes by using `dev-requirements` while AWS Glue Job Runs continue to use the older stable versions given in `requirements` (but this may change in a subsequent PR as we start introducing integration tests for Iceberg on AWS Glue).